server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -153,6 +137,11 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         return new DeleteResult(filesDeleted.get(), bytesDeleted.get());
     }
 
+    @Override
+    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
+        IOUtils.rm(blobNames.stream().map(path::resolve).toArray(Path[]::new));
```

Comment:
I think this is a fine replacement. `IOUtils.rm` checks for existence of a path before trying to delete it so it won't throw any `NoSuchFileException` unless there's concurrent modifications on the repo going on (in which case I think it's fine (and potentially better) to throw than to ignore the issue IMO)